cpp/src/io/parquet/page_hdr.cu

```diff
@@ -396,7 +396,7 @@ void __launch_bounds__(128) gpuDecodePageHeaders(ColumnChunkDesc* chunks,
     }
     num_values    = bs->ck.num_values;
     page_info     = chunk_pages ? chunk_pages[chunk].pages : nullptr;
-    max_num_pages = page_info ? bs->ck.max_num_pages : 0;
+    max_num_pages = page_info ? (bs->ck.num_data_pages + bs->ck.num_dict_pages) : 0;
```

Comment:
Why do we have this?

Reply:
`max_num_pages` was being computed on the host and stored in the chunk struct, but it was only ever used here, and we already have everything we need to compute it.  So I just removed it from the struct and compute it on the spot.